lib/ivis_opengl/piefunc.cpp

```diff
@@ -85,33 +85,18 @@ void pie_DrawViewingWindow(const glm::mat4 &modelViewProjectionMatrix)
 	radarViewGfx[1]->draw(modelViewProjectionMatrix);
 }
 
-namespace
-{
-	struct glBufferWrapper
-	{
-		GLuint id;
-		glBufferWrapper()
-		{
-			glGenBuffers(1, &id);
-		}
-
-		~glBufferWrapper()
-		{
-			glDeleteBuffers(1, &id);
-		}
-	};
-}
-
 void pie_TransColouredTriangle(const std::array<Vector3f, 3> &vrt, PIELIGHT c, const glm::mat4 &modelViewMatrix)
 {
 	pie_SetTexturePage(TEXPAGE_NONE);
 	pie_SetRendMode(REND_ADDITIVE);
 	glm::vec4 color(c.byte.r / 255.f, c.byte.g / 255.f, c.byte.b / 255.f, 128.f / 255.f);
 	const auto &program = pie_ActivateShader(SHADER_GENERIC_COLOR, pie_PerspectiveGet() * modelViewMatrix, color);
 
-	static glBufferWrapper buffer;
-	glBindBuffer(GL_ARRAY_BUFFER, buffer.id);
-	glBufferData(GL_ARRAY_BUFFER, 3 * sizeof(Vector3f), vrt.data(), GL_STREAM_DRAW);
+	static gfx_api::buffer* buffer;
+	if (buffer)
+		delete buffer;
+	buffer = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, 3 * sizeof(Vector3f), gfx_api::context::buffer_storage_hint::stream_draw);
```

Comment:
This is not equivalent code to the original code, and will introduce a performance regression. You are creating and deleting the buffer every time through. There is overhead to that.